src/pybind/mgr/dashboard/controllers/host.py

```diff
@@ -173,6 +175,7 @@ def populate_service_instances(hostname, services):
     return [{'type': k, 'count': v} for k, v in Counter(services).items()]
 
 
+@ttl_cache(30)
```

Comment:
I like this. Probably we should _immediately_ open a PR adding a TTL cache to all the list-like endpoints prone to scale up with the cluster size (hosts, devices, OSDs, daemons, alerts, devices, RGW buckets, etc.). This measure alone would automatically harden the Ceph Dashboard back-end against (intentional or accidental) DoS.

That said, I'd avoid hardcoding any limits if we can auto-tune this. My suggestion is to extend the `ttl_cache`, like this:
```py
@ttl_cache(min_ttl=5, multiplier=.5)
```
* **min_ttl**: this would set the lower hard threshold (probably 5 secs, since that's the shortest period the UI polls the back-end).
* **multiplier**: instead of a fixed ttl, I suggest that we measure how long it takes for the back-end data to be retrieved and we set the effective TTL to a multiple of that delay (an optimal `multiplier` might be 0.5-2, we would need to tweak this empirically once).
 
In this way, the TTL caching responds to the current conditions of the cluster/network and the complexity/cost of each call.